bofire/strategies/predictives/enting.py

```diff
@@ -0,0 +1,317 @@
+        self._problem_config: ProblemConfig = cfg[0]
+        self._model_pyo: pyo.ConcreteModel = cfg[1]
+
+    @property
+    def input_preprocessing_specs(self):
+        return {}
+
+    def _postprocess_candidate(self, candidate: List) -> pd.DataFrame:
+        """Converts a single candidate to a pandas Dataframe with prediction.
+
+        Args:
+            candidate (List): List containing the features of the candidate.
+
+        Returns:
+            pd.DataFrame: Dataframe with candidate.
+        """
+        keys = [feat.name for feat in self._problem_config.feat_list]
+        df_candidate = pd.DataFrame(
+            data=[candidate],
+            columns=keys,
+        )
+
+        preds = self.predict(df_candidate)
+
+        return pd.concat((df_candidate, preds), axis=1)
+
+    def _fantasy_as_experiment(self, candidates: pd.DataFrame):
+        """Fit the model with fantasy candidates.
+
+        The Enting strategy generates a globally optimal candidate. Therefore,
+        to generate batch proposals, we sequentially generate 'fantasy' observations
+        of the candidate, by adding a multiple of the standard deviation to the
+        mean prediction. This behaviour is defined by the `kappa_fantasy` parameter.
+
+        Args:
+            candidates (pd.DataFrame): The candidate(s) to make a fantasy observation for.
+        """
+        kappa = self._kappa_fantasy
+        # overestimate for minimisation, underestimate for maximisation
+        signs = {
+            output.key: -1 if isinstance(output.objective, MaximizeObjective) else 1
+            for output in self.domain.outputs.get()
+        }
+        as_experiment = candidates.assign(
+            **{
+                key: candidates[f"{key}_pred"]
+                + kappa * signs[key] * candidates[f"{key}_sd"]
+                for key in self.domain.outputs.get_keys()
+            },
+            valid_y=True,
+        )
+
+        return as_experiment
+
+    def _ask(self, candidate_count: PositiveInt = 1) -> pd.DataFrame:
+        """Generates candidates.
+
+        If `candidate_count == 1`, then the globally optimal solution is returned.
+        If `candidate_count > 1`, then we use fantasy observations to make sequential
+        proposals. Note that since this sequentially generates candidates, it is
+        much faster to generate a batch in a single function call, such that each candidate
+        is only predicted once.
+
+        Args:
+            candidate_count (PositiveInt, optional): Number of candidates to be generated. Defaults to 1.
+
+        Returns:
+            pd.DataFrame: DataFrame with a candidates.
+        """
+        # First, fit the model on fantasies generated for any pending candidates
+        # This ensures that new points are far from pending candidates
+        real_experiments = self.experiments
+        if self.candidates is not None:
+            for i in range(len(self.candidates)):
```

Comment:
It is possible, however it wouldn't be 'correct', I believe. Let's say we have candidates $(x_1, ..., x_5)$. When we initially generated these, we first generated $x_1$, then created a 'fantasy observation' of this point, and used that to generate $x_2$, and so on. This means that $x_5$ is conditional on the model **fitted to the fantasy observations of $(x_1, ..., x_4)$.** If we generated the fantasies at the same time, then the predictions would be different, which means $x_5$ may no longer be optimal as the fantasy observations of the other points would be different.

Ultimately this problem arises because ENTMOOT can only produce single, globally optimal candidates. There is no support for batch proposals in ENTMOOT.

If doesn't make sense, or you think the difference is too subtle, please let me know - happy to carry on this discussion!